runtime/oti/j9javaaccessflags.h

```diff
@@ -157,9 +157,7 @@
 #define J9StaticFieldRefBoolean 0x8
 #define J9StaticFieldRefPutResolved 0x10
 #define J9StaticFieldRefFinal 0x20
-#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
```

Comment:
As this ifdef is removed, can you add a comment above this flag saying that `J9StaticFieldIsNullRestricted` could only be set in Valhalla builds.

Reply:
Please explain the motivation for making this declaration unconditional (I only see uses where `defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)` is true).

Reply:
Wouldn't it not compile in the LinearDumper case? Or does DDR not pay attention to these?

Reply:
No, it should compile: DDR benefits from the declaration of `J9StaticFieldIsNullRestricted` in `CompatibilityConstants29.dat`.

Reply:
Oh right, I'll put those back then.